rt/transports/http/src/main/java/org/apache/cxf/transport/http/HttpClientHTTPConduit.java

```diff
@@ -349,13 +348,6 @@ protected void setupConnection(Message message, Address address, HTTPClientPolic
             return;
         }
 
-        if (sslURL != null && isSslTargetDifferent(sslURL, uri)) {
```

Comment:
@CantosSong thanks for the fix, but I lack some context here why this reset on HTTPS URL was even introduced. I would suggest we play safe here and introduce the contextual property that would do the reset if needed:

``
final boolean resetClientNeeded = MessageUtils.getContextualBoolean(message, RESET_CLIENT_ON_HTTPS_URL_CHANGE);
....
```